lib/collection/src/shards/local_shard/mod.rs

```diff
@@ -741,6 +757,28 @@ impl LocalShard {
         Ok(())
     }
 
+    /// Apply shard's strict mode configuration update
+    /// - Update read and write rate limiters
+    pub async fn on_strict_mode_config_update(&self) {
+        let config = self.collection_config.read().await;
+
+        if let Some(strict_mode_config) = &config.strict_mode_config {
```

Comment:
Would it be worth it to create a function that builds and returns the new `RateLimiter`s. We'd be able to use it in the constructor above, and here, sharing code.